src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -1987,13 +1977,28 @@ private object GetMessageJson(JsonRpcMessage message)
             }
         }
 
-#pragma warning disable AvoidAsyncSuffix // Avoid Async suffix
-        private ValueTask TransmitAsync(JsonRpcMessage message, CancellationToken cancellationToken)
-#pragma warning restore AvoidAsyncSuffix // Avoid Async suffix
+        private async ValueTask TransmitAsync(JsonRpcMessage message, CancellationToken cancellationToken)
         {
-            this.TraceMessageSent(message);
+            try
+            {
+                this.TraceMessageSent(message);
+                await this.MessageHandler.WriteAsync(message, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception exception)
+            {
+                if ((bool)(this.MessageHandler as IDisposableObservable)?.IsDisposed)
```

Comment:
This will throw if `MessageHandler` does not implement `IDisposableObservable` because you can't cast a `null` to `bool`. 
Instead, try this syntax:

```cs
if ((this.MessageHandler as IDisposableObservable)?.IsDisposed ?? false)
```